### PR TITLE
Do not allow reopening of closed events

### LINKED ISSUE #188: User should not be able to modify objects in closed state
This is a inconsistency between zino1 and zino2.
in zino1, when a case is in closed state it is complely locked for editing.

In zino2 when a ticket is in closed state the user should not be able to modify it. eg. adding history.
There could be a discussion on if it should be possible to reopen a case at all. 
There are instances where reopening a case could be beneficial, eg when closing a case by accident eg. by a mass-update.
But this again could make things like triggering an external action when closing cases more difficult as the close action could be done multiple times on a case object.

### PR DESCRIPTION
**Pull request by johannaengland**
Might close #188 (depending on if changing closed events should be allowed or not).

### CODE REVIEW

File: src/zino/statemodels.py
Comment (stveit) [inline src/zino/statemodels.py:265]: I would  put the ClosedEventError check further upp, above `old_state, self.state = self.state, new_state`. It looks like `self.state` gets set to `new_state`, and then afterwards you raise the ClosedEventError.

Comment (stveit) [inline src/zino/statemodels.py:265]: I would  put the ClosedEventError check further upp, above `old_state, self.state = self.state, new_state`. It looks like `self.state` gets set to `new_state`, and then afterwards you raise the ClosedEventError. I guess the test works since this doesnt get Commited (?), but would feel cleaner to just not update `self.state` at all

**Review by stveit — APPROVED** [review]
Looks fine to me

**Comment by johannaengland** [comment]
I would like to wait for a review from @lunkwill42 before merging, mainly for the design decision if this is enough to close #188.

**Review by lunkwill42 — APPROVED** [review]
Looks a-ok to me! Thanks!

**Pull request merged**
